src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,146 @@
 {
     using System;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Management.Automation;
-    using PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+    using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+    using Newtonsoft.Json;
 
     internal class PowerShellServices : IPowerShellServices
     {
-        private const string SetFunctionInvocationContextCommand =
-            "Microsoft.Azure.Functions.PowerShellWorker\\Set-FunctionInvocationContext";
+        private readonly string SetFunctionInvocationContextCommand;
+        private const string ExternalDurableSDKName = "DurableSDK";
+        private const string InternalDurableSDKName = "Microsoft.Azure.Functions.PowerShellWorker";
 
         private readonly PowerShell _pwsh;
-        private bool _hasSetOrchestrationContext = false;
+        private bool hasInitializedDurableFunctions = false;
+        private readonly bool hasExternalDurableSDK = false;
 
         public PowerShellServices(PowerShell pwsh)
         {
+            // We attempt to import the external SDK upon construction of the PowerShellServices object.
+            // We maintain the boolean member hasExternalDurableSDK in this object rather than
+            // DurableController because the expected input and functionality of SetFunctionInvocationContextCommand
+            // may differ between the internal and external implementations.
+
+            try
+            {
+                pwsh.AddCommand(Utils.ImportModuleCmdletInfo)
+                    .AddParameter("Name", ExternalDurableSDKName)
+                    .AddParameter("ErrorAction", ActionPreference.Stop)
+                    .InvokeAndClearCommands();
+                hasExternalDurableSDK = true;
+            }
+            catch (Exception e)
+            {
+                // Check to see if ExternalDurableSDK is among the modules imported or
+                // available to be imported: if it is, then something went wrong with
+                // the Import-Module statement and we should throw an Exception.
+                // Otherwise, we use the InternalDurableSDK
+                var availableModules = pwsh.AddCommand(Utils.GetModuleCmdletInfo)
+                    .AddParameter("Name", ExternalDurableSDKName)
+                    .InvokeAndClearCommands<PSModuleInfo>();
+                if (availableModules.Count() > 0)
+                {
+                    // TODO: evaluate if there is a better error message or exception type to be throwing.
+                    // Ideally, this should never happen.
+                    throw new InvalidOperationException("The external Durable SDK was detected, but unable to be imported.", e);
+                }
+                hasExternalDurableSDK = false;
+            }
+
+            if (hasExternalDurableSDK)
```

Comment:
Do we log anywhere what Durable SDK (internal or external) version the PowerShell worker is using?

Reply:
I added a log for this in the PowerShellManager.InvokeFunction method